internals/daemon/api_notices_test.go

```diff
@@ -205,18 +258,391 @@ func (s *apiSuite) TestNoticesFilterInvalidTypes(c *C) {
 	c.Assert(notices, HasLen, 0)
 }
 
+func (s *apiSuite) TestNoticesUserIDsRootDefault(c *C) {
+	s.daemon(c)
+	restore := mockSysGetuid(0)
+	defer restore()
+
+	st := s.d.overlord.State()
+	st.Lock()
+	addNotice(c, st, 0, state.ChangeUpdateNotice, "123", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 1000, state.CustomNotice, "a.b/x", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 123, state.CustomNotice, "a.b/y", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 0, state.WarningNotice, "danger", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	addNotice(c, st, 1000, state.ChangeUpdateNotice, "456", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	addNotice(c, st, 123, state.ChangeUpdateNotice, "789", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	st.Unlock()
+
+	noticesCmd := apiCmd("/v1/notices")
+
+	// Test that root user sees all notices if no filter is specified
+	req, err := http.NewRequest("GET", "/v1/notices", nil)
+	c.Assert(err, IsNil)
+	req.RemoteAddr = "pid=100;uid=0;socket=;"
+	rsp, ok := noticesCmd.GET(noticesCmd, req, nil).(*resp)
+	c.Assert(ok, Equals, true)
+
+	c.Check(rsp.Type, Equals, ResponseTypeSync)
+	c.Check(rsp.Status, Equals, http.StatusOK)
+	notices, ok := rsp.Result.([]*state.Notice)
+	c.Assert(ok, Equals, true)
+	c.Assert(notices, HasLen, 6)
+	n := noticeToMap(c, notices[0])
+	c.Assert(n["user-id"], Equals, 0.0)
+	n = noticeToMap(c, notices[1])
+	c.Assert(n["user-id"], Equals, 1000.0)
+	n = noticeToMap(c, notices[2])
+	c.Assert(n["user-id"], Equals, 123.0)
+	n = noticeToMap(c, notices[3])
+	c.Assert(n["user-id"], Equals, 0.0)
+	n = noticeToMap(c, notices[4])
+	c.Assert(n["user-id"], Equals, 1000.0)
+	n = noticeToMap(c, notices[5])
+	c.Assert(n["user-id"], Equals, 123.0)
+}
+
+func (s *apiSuite) TestNoticesUserIDsRootFilter(c *C) {
+	s.daemon(c)
+	restore := mockSysGetuid(0)
+	defer restore()
+
+	st := s.d.overlord.State()
+	st.Lock()
+	addNotice(c, st, 0, state.ChangeUpdateNotice, "123", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 1000, state.CustomNotice, "a.b/x", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 123, state.CustomNotice, "a.b/y", nil)
+	time.Sleep(time.Microsecond)
+	addNotice(c, st, 0, state.WarningNotice, "danger", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	addNotice(c, st, 1000, state.ChangeUpdateNotice, "456", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	addNotice(c, st, 123, state.ChangeUpdateNotice, "789", &state.AddNoticeOptions{
+		Visibility: state.PublicNotice,
+	})
+	st.Unlock()
+
+	noticesCmd := apiCmd("/v1/notices")
+
+	// Test that root can filter on any user IDs
+	for _, testCase := range []struct {
+		filterUserIDs   []uint32
+		expectedUserIDs []float64
+	}{
+		{
+			[]uint32{0},
+			[]float64{0.0, 0.0},
+		},
```

Comment:
Let's use the terser bracing style as per the rest of the codebase, and add field names (but shorten them to `filter` and `expected`):

```go
{{
    filter: []uint32{0},
    expected: []float64{0.0, 0.0},
}, {
    filter: ...,
    expected: ...,
}, {
    ...
```

Similar below.